lib/ios-log.js

```diff
@@ -0,0 +1,203 @@
+
+    if ( ver < 5 ) {
+      logger.debug('Starting iOS6.* simulator log capture');
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', '/var/log/system.log']);
+      await this.finishStartingLogCapture();
+      return;
+    }
+
+    let logsPath;
+
+    if (ver >= 6) { 
+      logger.debug('Starting iOS 8.*/9.* simulator log capture');
+        if (_.isUndefined(this.simUdid)) {
+          logger.errorAndThrow('iOS 8*/9.* log capture requires a sim udid'); 
+        }
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'CoreSimulator', this.simUdid); 
+    } else {
+        logger.debug('Starting iOS7.* simulator log capture');
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'iOS Simulator', '7.*');
+    }
+
+    try {
+      if (logsPath.indexOf('*') >= 0) {
+        return; 
+      }
+      await mkdirp(logsPath);
+      await touch(path.resolve(logsPath, 'system.log'));
+      await fs.appendFile(path.resolve(logsPath, 'system.log'), 
+            'A new Simulator is about to start!'); 
+      let files;
+      try {
+        files = await glob(path.resolve(logsPath, 'system.log'));
+        if (files.length < 1) {
+          throw 'Could not start log capture';
+        }
+      } catch (e) {
+        logger.error(`Could not start log capture because no iOS ` +
+        `simulator logs could be found at ${logsPath}/system.log` +
+        `Logging will not be functional for this run`);
+      }
+
+      let lastModifiedLogPath = files[0];
+      let lastModifiedLogTime = await fs.stat(files[0]).mtime;
+      for (let file in files) {
+        let mtime = await fs.stat(file).mtime;
+        if (mtime > lastModifiedLogTime) {
+          lastModifiedLogPath = file;
+          lastModifiedLogTime = mtime;
+        }
+      }
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', lastModifiedLogPath]);
+      await this.finishStartingLogCapture();
+    } catch (err) {
+      logger.warn('System log capture failed.');
+      return;
+    }    
+  }  
+
+  async finishStartingLogCapture () {
+    if (!this.proc) {
+      logger.errorAndThrow('Could not capture device log');
+    }
+    this.startTimer = setTimeout(function () {
+      logger.errorAndThrow('Log capture did not start in a reasonable amount of time');
+    }, START_TIMEOUT);
+    let firstLine = true;
+
+    //this.proc.on --> start --> function startDetector
+    this.proc.on('output', (stdout, stderr) => {
+      if (stdout) {
+        if (firstLine) {
+          if (stdout.substr(-1,1) === '\n') {
+              this.onOutput();
+              firstLine = false; 
+          } 
+        } else {
+          this.logRow += stdout;
+          if (stdout.substr(-1,1) === '\n') {
+            this.onOutput('');
+            this.logRow = ''; 
+          }
+        }
+      } 
+      if (stderr) {
+        this.onOutput(' STDERR');
+      }
+    });
+
+    let sd = (stdout, stderr) => {
```

Comment:
pretty sure the startDetector also needs to return `true` whenever we actually do have a successful start condition. in our case, that would probably be anytime we have any output, so you can do:

```js
return stdout || stderr;
```

also, it should read:

```js
if (/execvp...
```

(space after `if`)